roofit/roofitcore/inc/RooAbsArg.h

```diff
@@ -610,7 +610,12 @@ class RooAbsArg : public TNamed, public RooPrintable {
   RefCountList_t _clientListShape; // subset of clients that requested shape dirty flag propagation
   RefCountList_t _clientListValue; // subset of clients that requested value dirty flag propagation
 
+  struct ProxyListCache {
+    std::vector<RooAbsProxy*> cache;
+    bool isDirty = true;
+  };
   RooRefArray _proxyList        ; // list of proxies
+  ProxyListCache _proxyListCache; //! cache of the list of proxies
```

Comment:
It would be nice to move this down somehow, since the `bool` will force the compiler to add padding bytes to have `_proxyList` aligned.
You can minimise the size of the class by putting this right before another bool.

Reply:
e.g. line 672

Reply:
... better right before `_valueDirty`.

Reply:
Wow, I had no clue about such things. Fixed as you suggested